src/components/TransactionDialogs/DelegationDialog/SelectValidators.tsx

```diff
@@ -0,0 +1,396 @@
+  Box,
+  DialogActions,
+  DialogContent,
+  IconButton,
+  InputAdornment,
+  TextField,
+  Typography,
+  Grid,
+  CircularProgress,
+  useTheme,
+  Slider,
+  Card,
+} from '@mui/material'
+import React from 'react'
+import Avatar from '../../Element/avatar'
+import { useTranslation } from 'react-i18next'
+import useIconProps from '../../../misc/useIconProps'
+import DropdownIcon from '../../../components/svg/DropdownIcon'
+import { ReactComponent as RemoveIcon } from '../../../assets/images/icons/icon_clear.svg'
+import MemoInput from '../../../components/MemoInput'
+import { Loadable, useRecoilValueLoadable } from 'recoil'
+import { randomizedValidatorsState } from '../../../recoil/validators'
+
+interface SelectValidatorsProps {
+  onConfirm(delegations: Array<{ amount: number; validator: Validator }>, memo: string): void
+  delegations: Array<{ amount: number; validator: Validator }>
+  //   price: number
+  validators: Loadable<Validator[]>
+  amount: number
+  denom: string
+  loading: boolean
+  account: AccountDetail
+  validatorsMap: { [name: string]: Validator }
+}
+
+const SelectValidators = ({
+  account,
+  onConfirm,
+  delegations: defaultDelegations,
+  validators,
+  amount,
+  denom,
+  loading,
+  validatorsMap,
+}: SelectValidatorsProps) => {
+  const { t } = useTranslation('common')
+  const iconProps = useIconProps()
+  const [consent, setConsent] = React.useState(true)
+  const [memo, setMemo] = React.useState('')
+  const [delegations, setDelegations] = React.useState<
+    Array<{ amount: string; validator: any; percentage: string; showSlider: boolean }>
+  >(
+    defaultDelegations
+      ? defaultDelegations.map((d) => ({
+          amount: d.amount.toString(),
+          validator: d.validator,
+          percentage: ((100 * d.amount) / amount).toFixed(2),
+          showSlider: false,
+        }))
+      : [{ amount: amount.toString(), validator: {}, percentage: '100', showSlider: false }]
+  )
+
+  const randomizedValidators = useRecoilValueLoadable(
+    randomizedValidatorsState({
+      chainId: account.chain,
+    })
+  )
+
+  React.useMemo(() => {
+    setDelegations((d) =>
+      d.map((a, j) =>
+        j < delegations.length - 1
+          ? {
+              ...a,
+              percentage: String(((1 / delegations.length) * 100).toFixed(2)) || '',
+              amount: String(amount * (1 / delegations.length)),
+            }
+          : {
+              ...a,
+              percentage:
+                String(
+                  (100 - (1 / delegations.length) * (delegations.length - 1) * 100).toFixed(2)
+                ) || '',
+              amount: String(amount * (1 - (1 / delegations.length) * (delegations.length - 1))),
+            }
+      )
+    )
+  }, [delegations.length])
+
+  const totalAmount = React.useMemo(
+    () => delegations.map((v) => Number(v.amount)).reduce((a, b) => a + b, 0),
+    [delegations]
+  )
+
+  return (
+    <form
+      noValidate
+      onSubmit={(e) => {
```

Comment:
This `onSubmit` function can be turned into a `useCallback`.

Remember to set the dependencies accordingly